core/changedetector/proto_change_detector_service_test.go

```diff
@@ -0,0 +1,268 @@
+package changedetector_test
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/goto/stencil/core/changedetector"
+	"github.com/goto/stencil/formats/protobuf"
+	mocks2 "github.com/goto/stencil/pkg/newrelic/mocks"
+	stencilv1beta2 "github.com/goto/stencil/proto/gotocompany/stencil/v1beta1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"io/ioutil"
+	"log"
+	"path/filepath"
+	"testing"
+)
+
+func getSvc() (*changedetector.Service, *mocks2.NewRelic) {
+	newRelic := &mocks2.NewRelic{}
+	svc := changedetector.NewService(newRelic)
+	return svc, newRelic
+}
+
+func TestIdentifySchemaChange(t *testing.T) {
+	ctx := context.Background()
+
+	t.Run("should return error if previous schema data is nil", func(t *testing.T) {
+		svc, newRelic := getSvc()
+		request := &changedetector.ChangeRequest{
+			NamespaceName: "testNamespace",
+			SchemaName:    "testSchemaName",
+			Version:       int32(1),
+			NewData:       []byte("b"),
+		}
+		var called bool
+		newRelic.On("StartGenericSegment", mock.Anything, "Identify Schema Change").Return(func() { called = true })
```

Comment:
Can we have beforeTest method which statements are common to all the below sub-tests?

Reply:
added init method and moved request declaration